### ISSUE TITLE
upgrade failed from 0.59.8 to 0.59.9

### ISSUE THREAD

**Issue by KingAmo** [react-native-community/rn-diff-purge#17]
# Bug
```
$ react-native upgrade
info No version passed. Fetching latest...
info Fetching diff between v0.59.8 and v0.59.9...
info Applying diff (excluding: package.json)...
tmp-upgrade-rn.patch:68: trailing whitespace.
@rem
tmp-upgrade-rn.patch:69: trailing whitespace.
@rem Copyright 2015 the original author or authors.
tmp-upgrade-rn.patch:70: trailing whitespace.
@rem
tmp-upgrade-rn.patch:71: trailing whitespace.
@rem Licensed under the Apache License, Version 2.0 (the "License");
tmp-upgrade-rn.patch:72: trailing whitespace.
@rem you may not use this file except in compliance with the License.
error: cannot apply binary patch to 'android/gradle/wrapper/gradle-wrapper.jar' without full index line
Falling back to three-way merge...
error: cannot apply binary patch to 'android/gradle/wrapper/gradle-wrapper.jar' without full index line
error: android/gradle/wrapper/gradle-wrapper.jar: patch does not apply
error Automatically applying diff failed
error Patch failed to apply for unknown reason. Please fall back to manual way of upgrading
info You may find these resources helpful:
• Release notes: https://github.com/facebook/react-native/releases/tag/v0.59.9
• Comparison between versions: https://github.com/react-native-community/rn-diff-purge/compare/version/0.59.8..version/0.59.9
• Git diff: https://github.com/react-native-community/rn-diff-purge/compare/version/0.59.8..version/0.59.9.diff
error Upgrade failed. Please see the messages above for details. Run CLI with --verbose flag for more details.
```

## Environment info

React native info output:

```bash
React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
      Memory: 139.79 MB / 8.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 12.3.1 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 28
        Build Tools: 28.0.3
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3
      react-native: 0.59.8 => 0.59.8
    npmGlobalPackages:
      react-native-cli: 2.0.1
```

Library version: x.x.x

## Steps To Reproduce
run `react-native upgrade`

Describe what you expected to happen:

1.  successful upgrade from 0.59.8 to 0.59.9

**Comment by pvinis** [comment]
Unfortunately it says that the binary .jar file can't be replaced. I suggest you use https://github.com/react-native-community/rn-diff-purge/compare/release/0.59.8..release/0.59.9 and do the upgrade manually, and for the jar, just init a new project and copy it over. I will make an issue in the cli to check how to handle binary files.

**Comment by Aung-Myint-Thein** [comment]
According to this comment, if you are running 59.8, upgrading to 59.9 should be only a matter of changing the version number in package.json. https://github.com/react-native-community/releases/issues/124#issuecomment-499154739

And the changelog also said that 

> This is a patch fix release addressing a couple ScrollView regressions, and "future-proof" RN 59 from crashes caused by upgrading Gradle (now can support up to 5.4.1 & 3.4.0 for the plugin) and Xcode 11 Beta 1. You can upgrade to this version without upgrading your tooling.

https://github.com/facebook/react-native/releases/tag/v0.59.9

I tried to upgrade the Gradle and tried to run. It is fine but when I tried to get the APK, a lot of issues (mainly in gradle and bundle etc) coming out. So, I am just changing the version number in package.json. However, the risk is that when I upgrade to RN 0.60, I might miss out some of the changes in gradle files. So, if anyone came here with this issue, you can use this as an alternative but please know the risk.

**Comment by Aung-Myint-Thein** [comment]
And in the https://github.com/react-native-community/rn-diff-purge/compare/release/0.59.8..release/0.59.9 I saw there are some changes in JVM arguments used for the daemon process. Do we really need to change them? Look like it is giving heap out of memory issues. I could be wrong...

**Comment by thymikee** [comment]
Fixed in `@react-native-community/cli@1.9.10`, make sure to re-install it.
